src/acl.c

```diff
@@ -1922,19 +1922,9 @@ void ACLKillPubsubClientsIfNeeded(user *new, user *original) {
         kill = 0;
 
         if (c->user == original && getClientType(c) == CLIENT_TYPE_PUBSUB) {
-            /* Check for pattern violations. */
-            listRewind(c->pubsub_patterns,&lpi);
-            while (!kill && ((lpn = listNext(&lpi)) != NULL)) {
-
-                o = lpn->value;
-                int res = ACLCheckChannelAgainstList(upcoming, o->ptr, sdslen(o->ptr), 1);
-                kill = (res == ACL_DENIED_CHANNEL);
-            }
-            /* Check for channel violations. */
```

Comment:
i'm not sure why you moved it, but i don't mind.
however, now we seem to have a silly check for `if (!kill)` that's always true.
and the code inside unconditionally creates dictIterator 3 times.
i wouldn't want to give up the silly `if` since it'll mess up the indentation and blame log, but what if we have an `if (!kill)` before each one of the iterations, instead of the one that wraps all 3?
it would mean the majority of the code retains it's indentation level, and we possibly save time on creating useless iterators.
i suppose that was the original purpose (before sharded channels were added)

Reply:
the reson i moved it is because: I will modify basically all lines anyway, i adjust its position by the way